### DISCUSSION TITLE
Is it production ready?

### DISCUSSION THREAD

**Discussion by laith43d** [piccolo-orm/piccolo#1325] [General]
Hi,

I want to congratulate you on this successful project.

I came across Piccolo a few weeks ago. I use Django-Ninja to create APIs, moved from DRF for many reasons beyond this comment.

Django-Ninja benefits FastAPI with Django ORM and Admin; however, I believe Piccolo represents a huge potential to move back to FastAPI. I already tried it on one of my projects and absolutely loved it.

Despite the feature limitation, like an obvious and automatic way to implement M2M, it has all the features required by most of my projects.

My question is: is Piccolo stable enough for production on mid to large-scale projects? pet projects don't reveal the beast inside :) and I very much like to try the performance gains of FastAPI combined with Piccolo!

Best of luck

**Comment by dantownsend** [comment]
@laith43d Thanks for the kind words.

As you mention, Piccolo is missing some features (mostly M2M columns, but I hope to add these soon).

I do think Piccolo is production ready. I use it for all of my own professional work, so am 100% committed to it being stable, and continuing to add features.

We rarely make backwards incompatible changes, and when we do they are very minor.

I can't promise there aren't some minor bugs we haven't discovered yet, but we release very frequently, so issues do get fixed quickly.

**Comment by dashsatish** [comment]
Considering the basic features with type annotations and other conveniences. I feel Piccolo to be a best combination with FastAPI. And we have one micro-service in production running fluently for over 7 months now (At the time of writing this comment)

@dantownsend and others did an awesome job in clarifying our questions and fixing basic issues. Kudos to all the volunteers here. It's a great framework, enhancements and a few other awesome features are in the road map i believe.
